src/core/utils/channel_manager.cpp

```diff
@@ -112,7 +112,7 @@ void ChannelManager::RequestNetworkChannelChange(uint8_t aChannel)
 #if OPENTHREAD_CONFIG_CHANNEL_MANAGER_CSL_CHANNEL_SELECT_ENABLE
 void ChannelManager::ChangeCslChannel(uint8_t aChannel)
 {
-    if (!(!Get<Mle::Mle>().IsRxOnWhenIdle() && Get<Mac::Mac>().IsCslEnabled()))
+    if (Get<Mle::Mle>().IsRxOnWhenIdle() || !Get<Mac::Mac>().IsCslEnabled())
```

Comment:
Seems it's still changing the boolean expressions.

Reply:
This one is changed last time and I didn't change it back. I've changed it back now.